### ISSUE TITLE
Wrong items are moved in advanced inventory screen

### ISSUE THREAD

**Issue by Inglonias** [CleverRaven/Cataclysm-DDA#40109]
# Describe the bug
Advanced inventory cannot be used to transfer items properly

# Steps To Reproduce
1. Load the attached save file [Literal Actual 2020 Simulator.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4572041/Literal.Actual.2020.Simulator.zip)
2. Open the advanced inventory screen
3. Try to move the cellphone to the tile south of you.
4. My character has taken their pants off for some reason.

# Expected behavior
The cellphone is moved.

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-1662-g686104e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    No Rail Stations [No_Rail_Stations],
    No Fungal Monsters [No_Fungi],
    Disable NPC Needs [no_npc_food]
]

**Comment by Orange01gaming** [comment]
I had the same thing happen when I tried to move a jain I had in my survivor harness. Other items in regular pockets were able to move just fine.

**Comment by masseymo** [comment]
I have also experienced this. I tried to move a phone and my character removes his socks.
I tried to move a flashlight and it gave an error
"src/character.cpp:2447 [item Character::i_rem(const item *)] did not found item none to remove it!"
the flashlight didn't move and it created a 'none' object instead.
It seems like the advanced inventory is basically unusable right now.

Mac osx
Game Version: 0.E-4211-g7fcce2c
